documentation/tutorial.md

```diff
@@ -68,3 +75,762 @@ tutorial overlord    localhost/localhost  2.9.37   unsupported  23:20:53Z
+
+App                        Version  Status   Scale  Charm                      Channel   Rev  Exposed  Message
+mongodb                             active       2  mongodb                    dpe/edge   96  no       Replica set primary
+tls-certificates-operator           blocked      1  tls-certificates-operator  edge       16  no       Configuration options missing: ['certificate', 'ca-certificate']
+
+Unit                          Workload  Agent  Machine  Public address  Ports      Message
+mongodb/0*                    active    idle   0        10.23.62.156    27017/tcp  Replica set primary
+mongodb/1                     active    idle   1        10.23.62.55     27017/tcp  Replica set secondary
+tls-certificates-operator/0*  blocked   idle   3        10.23.62.8                 Configuration options missing: ['certificate', 'ca-certificate']
+
+Machine  State    Address       Inst id        Series  AZ  Message
+0        started  10.23.62.156  juju-d35d30-0  focal       Running
+1        started  10.23.62.55   juju-d35d30-1  focal       Running
+3        started  10.23.62.8    juju-d35d30-3  focal       Running
+```
+
+Now we can configure the TLS certificates. Configure the  `tls-certificates-operator` to use self signed certificates:
+```shell
+juju config tls-certificates-operator generate-self-signed-certificates="true" ca-common-name="Tutorial CA" 
+```
+*Note: this tutorial uses (self signed certificates)[https://en.wikipedia.org/wiki/Self-signed_certificate]; self signed certificates should not be used in a production cluster.*
+
+### Enable TLS
+After configuring the certificates `watch -c juju status --color` will show the status of `tls-certificates-operator` as active. To enable TLS on Charmed MongoDB, relate the two applications:
+```shell
+juju relate tls-certificates-operator mongodb
+```
+
+### Connect to MongoDB with TLS
+Like before, generate and save the URI that is used to connect to MongoDB:
+```
+export URI=mongodb://$DB_USERNAME:$DB_PASSWORD@$HOST_IP$DB_NAME?replicaSet=$REPL_SET_NAME
+echo $URI
+```
+Now ssh into `mongodb/0`:
+```
+juju ssh mongodb/0
+```
+After `ssh`ing into `mongodb/0`, we are now in the unit that is hosting Charmed MongoDB. Once TLS has been enabled we will need to change how we connect to MongoDB. Specifically we will need to specify the TLS CA file along with the TLS Certificate file. These are are on the units hosting the Charmed MongoDB application in the folder `/etc/mongodb/`. If you enter: `ls /etc/mongodb/external*` you should see the external certificate file and the external CA file:
+```shell
+/etc/mongodb/external-ca.crt  /etc/mongodb/external-cert.pem
+```
+
+As before, we will connect to MongoDB via the saved MongoDB URI. Connect using the saved URI and the following TLS options:
+```shell
+mongosh mongodb://$DB_USERNAME:$DB_PASSWORD@$HOST_IP/$DB_NAME?replicaSet=$REPL_SET_NAME --tls --tlsCAFile /etc/mongodb/external-ca.crt --tlsCertificateKeyFile /etc/mongodb/external-cert.pem
```

Comment:
I got blocked here - I'm not sure exactly what happened, except that at one point I was exiting out of mongosh etc and closed my terminal window. I don't know if this caused the problem, but this is what I got when I tried to run this command. 

![image](https://user-images.githubusercontent.com/116022293/211789707-d105db54-5a10-4705-a214-fd0af6747f25.png)

Reply:
Yes! You've found an edge case to the tutorial. So the URI is generated by environment variables. If you exit the terminal where these environment variables are defined then those definitions go away. Which is why when you logged back in and did:
<img width="704" alt="Screenshot 2023-01-11 at 12 49 21" src="https://user-images.githubusercontent.com/32723809/211810482-4a4a9926-c570-49eb-909b-bd1d9b18f860.png">
you have an URI without any content.

If you try to use a URI without any content you get this error message:
<img width="581" alt="Screenshot 2023-01-11 at 12 51 04" src="https://user-images.githubusercontent.com/32723809/211810823-e29a4526-6cdb-47ff-a164-dd7f407430b3.png">

I can put a disclaimer to this effect when generating the URI if you think that is a good idea :)

Reply:
It might be worth adding a small note somewhere (maybe by the first occasion where we have to type exit more than once - I exited out of the terminal by accident which could feasibly happen to a user as well).

Reply:
Yes! I think that is a much better place for this! Thanks for the suggestion!

Reply:
let me know what you think, changes are on line 268 :)